pkg/bgpv1/manager/reconcilerv2/crd_status.go

```diff
@@ -0,0 +1,277 @@
+					r.Logger.WithError(err).Error("Failed to update CRD status")
+				}
+
+			case <-ctx.Done():
+				r.Logger.Debug("CRD status update job stopped")
+				return nil
+			}
+		}
+	}))
+
+	return StatusReconcilerOut{
+		Reconciler: r,
+	}
+}
+
+func (r *StatusReconciler) Name() string {
+	return "CRDStatus"
+}
+
+func (r *StatusReconciler) Priority() int {
+	return 50
+}
+
+func (r *StatusReconciler) Reconcile(ctx context.Context, params StateReconcileParams) error {
+	r.Lock()
+	defer r.Unlock()
+
+	r.Logger.WithFields(logrus.Fields{
+		types.InstanceLogField: params.UpdatedInstance.Config.Name,
+	}).Debug("Reconciling CRD status")
+
+	// do not reconcile if not in BGPv2 mode
+	if params.ConfigMode.Get() != mode.BGPv2 {
+		return nil
+	}
+
+	current := &v2alpha1.CiliumBGPNodeStatus{}
+	if r.desiredStatus != nil {
+		current = r.desiredStatus.DeepCopy()
+	}
+
+	if params.UpdatedInstance != nil {
+		// get updated status for the instance
+		instanceStatus, err := r.getInstanceStatus(ctx, params.UpdatedInstance)
+		if err != nil {
+			return err
+		}
+
+		found := false
+		for idx, instance := range current.BGPInstances {
+			if instance.Name == instanceStatus.Name {
+				current.BGPInstances[idx] = *instanceStatus
+				found = true
+				break
+			}
+		}
+		if !found {
+			current.BGPInstances = append(current.BGPInstances, *instanceStatus)
+		}
+	}
+
+	if params.DeletedInstance != "" {
+		// remove instance from status
+		for idx, instance := range current.BGPInstances {
+			if instance.Name == params.DeletedInstance {
+				current.BGPInstances = append(current.BGPInstances[:idx], current.BGPInstances[idx+1:]...)
+				break
+			}
+		}
+	}
+
+	r.desiredStatus = current
+	return nil
+}
+
+func (r *StatusReconciler) getInstanceStatus(ctx context.Context, instance *instance.BGPInstance) (*v2alpha1.CiliumBGPNodeInstanceStatus, error) {
+	res := &v2alpha1.CiliumBGPNodeInstanceStatus{
+		Name:     instance.Config.Name,
+		LocalASN: instance.Config.LocalASN,
+	}
+
+	// get peer status
+	peers, err := instance.Router.GetPeerState(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, configuredPeers := range instance.Config.Peers {
+		if configuredPeers.PeerASN == nil || configuredPeers.PeerAddress == nil {
+			continue
+		}
+
+		peerStatus := v2alpha1.CiliumBGPNodePeerStatus{
+			Name:        configuredPeers.Name,
+			PeerAddress: *configuredPeers.PeerAddress,
+			PeerASN:     configuredPeers.PeerASN,
+		}
+
+		for _, runningPeerState := range peers.Peers {
+			if runningPeerState.PeerAddress == *configuredPeers.PeerAddress && runningPeerState.PeerAsn == *configuredPeers.PeerASN {
+				peerStatus.PeeringState = ptr.To[string](runningPeerState.SessionState)
+
+				// Update timestamp
+				peerStatus.LastUpdateTimestamp = ptr.To[string](time.Now().Format(time.RFC3339))
+
+				// Uptime is in nanoseconds, round to seconds
+				peerStatus.Uptime = ptr.To[string](time.Duration(runningPeerState.UptimeNanoseconds).Round(time.Second).String())
```

Comment:
I wouldn't be concerned about the uptime not being accurate too much, but what I mainly meant was that this comparison:
```
r.desiredStatus.DeepEqual(r.runningStatus)
```
will always result as non-equal if we keep a time in nanoseconds (that keeps counting every second) in the status. If it was a timestamp of the last transition instead, the value does not change until the peering state changes.

Reply:
I am not following this concern correctly. 

DesiredStatus is only updated when Reconcile is called, which happens whenever there is a state change ( peer or routing table ). Uptime in `desired status` is only set during this Reconcile. Once `running status` matches it and the BGP session stabilizes, we will not have a change to `uptime` in `desired status.`

Reply:
I see concerns about each state reconcile trigger; we will also update CRD since the timestamp will change.

Reply:
Hmm, this would require making non-backwards compatible change to CRD, since uptime is already part of the status field.

Reply:
> I see concerns that each state reconcile trigger; we will also update CRD since the uptime will change.

Is this a big deal, though? It just keeps the status field up-to-date.

Reply:
Ah, sorry I missed that DesiredStatus is only updated when Reconcile is called, so that's really not an issue.

I still think that exposing something like `EstablishedTime` (timestamp) instead of `Uptime` (duration) + `LastUpdateTimestamp` would be better. But also have nothing against keeping it in the original version if it does not cause more frequent status updates. 

Not sure if low cadence periodic update from gobgp is that necessary, I probably would not bother with that - but up to you.

nit: please note the [naming conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions) rule:
> The name of a field that specifies the time at which something occurs should be called somethingTime. Do not use stamp (e.g., creationTimestamp).

Reply:
Ah good point regarding non-compatible CRD change, but as status was never populated before, I think a change here it could be allowed?

Reply:
> as status was never populated before, I think a change here it could be allowed?

Yes, that's a possibility. I will update CRD in that case. 

Couple of changes
- Remove `uptime` and `last update timestamp`,  and introduce `established time.`
- Remove sum of route counts, only keep per afi/safi received and advertised route count.

Reply:
With the latest change, established time is published.

Sample output 

```
 kubectl get ciliumbgpnodeconfigs/bgpv2-cplane-dev-service-worker -o json | jq '.status'
{
  "bgpInstances": [
    {
      "localASN": 65001,
      "name": "65001",
      "peers": [
        {
          "establishedTime": "2024-06-27T17:30:27Z",
          "name": "65000",
          "peerASN": 65000,
          "peerAddress": "fd00:10:0:1::1",
          "peeringState": "established",
          "routeCount": [
            {
              "advertised": 9,
              "afi": "ipv4",
              "received": 8,
              "safi": "unicast"
            },
            {
              "advertised": 2,
              "afi": "ipv6",
              "received": 1,
              "safi": "unicast"
            }
          ],
          "timers": {
            "appliedHoldTimeSeconds": 90,
            "appliedKeepaliveSeconds": 30
          }
        }
      ]
    }
  ]
}
```

Reply:
awesome!